rebac-admin-backend/v1/core.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2024 Canonical Ltd.
+
+// Package v1 provides HTTP handlers that implement the ReBAC Admin OpenAPI spec.
+// This package delegates authorization and data manipulation to user-defined
+// "backend"s that implement the designated abstractions.
+package v1
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/canonical/identity-platform-admin-ui/rebac-admin-backend/v1/resources"
+)
+
+// ReBACAdminBackendParams contains references to user-defined implementation
+// of required abstractions, called "backend"s.
+type ReBACAdminBackendParams struct {
+	GroupsService       GroupsServiceBackend
+	GroupsAuthorization GroupsAuthorizationBackend
+}
+
+// ReBACAdminBackend represents the ReBAC admin backend as a whole package.
+type ReBACAdminBackend struct {
+	params  ReBACAdminBackendParams
+	service service
```

Comment:
like this it will be impossible to mock
I don't have the big picture but I guess this holds all the logic, if that's the case we need that as interface

Reply:
We don't need to mock this object (explained below), but, nevertheless, liked the idea in general. This `service` field already implements an auto-generated interface called `ServerInterface`, so we can just replace the type with that. 

About mocking `service`, note that it's the outmost layer that receives the requests (dispatched by the HTTP mux) and therefore we wouldn't normally want to mock it. Unless the code needs to invoke other endpoints in a loopback way (I mean not by invoking internal underlying methods, but going through the HTTP mux), which is not imaginable here.

Reply:
`ServerInterface` is like this:

```go
// ServerInterface represents all server handlers.
type ServerInterface interface {
	// List configured authentication providers.
	// (GET /authentication)
	GetAuthentication(w http.ResponseWriter, r *http.Request, params GetAuthenticationParams)
	// Configure a new authentication provider.
	// (POST /authentication)
	PostAuthentication(w http.ResponseWriter, r *http.Request)
	// Returns the list of supported identity providers.
	// (GET /authentication/providers)
	// ...
}
```